Use 'locate' command to find jboss-modules.jar

noahl opened the pull request:
This lets us scan the whole filesystem (or at least the locate
database) when it's been preindexed.

I've tested it on a test VM that does have jboss-modules.jar and locate installed, and one that does not have either. I've also tried using locate to find a file that doesn't exist, and I made sure that case is covered by unit tests.

Closes #291 .

Commit:
Use 'locate' command to find jboss-modules.jar

This lets us scan the whole filesystem (or at least the locate
database) when it's been preindexed.

chambridge commented on rho/facts.py:
```diff
@@ -168,6 +168,9 @@ def new_fact(fact_name, description, is_default=None,
          is_default=False)
+new_fact('jboss.eap.locate-jboss-modules-jar',
+         'Use locate to find jboss-modules.jar',
+         is_default=True)
```
Should this and the ones above be added to the JBOSS_FACTS category?

noahl replied:
Added locate to the JBOSS_FACTS category, thank you!

Everything in JBOSS_FACTS will be run when the user says `--facts jboss`, so I think we don't want to include the find-based commands in that. We should probably find a better name for the fact group, but I don't have one right now.

chambridge reviewed (commented):
Need to publish the fact docs update.

Commit:
Review fixes

Add locate to the 'jboss' fact group and update the fact docs. Thanks
to chambrid for the review.

noahl commented:
@chambridge published the fact doc update and added the new fact to JBOSS_FACTS. Thank you for the review!
